Hw04 auth

alicjapol opened the pull request.

alicjapol committed:
c

alicjapol committed:
commit

alicjapol committed:
Update contacts API to use new contact model

- Change express and other dependencies to their latest versions
- Import and use Joi for input validation
- Define a schema for contact input
- Implement handlers for GET /contacts, GET /contacts/:contactId, POST /contacts, DELETE /contacts/:contactId, and PUT /contacts/:contactId
- Import and use contactsOperations for database operations
- Remove unused middleware (logger and cors) from app.js

Add new dependencies: joi and nanoid
Update dependencies: express, cors, and morgan
Update models/contacts.js to use fs/promises and add new functions for database operations
Update package-lock.json with new dependencies and their versions

alicjapol committed:
pull rq

alicjapol committed:
simple express delated

alicjapol committed:
favourite component added

alicjapol committed:
crud impl

KamilMr commented on app.js:
```diff
@@ -1,25 +1,40 @@
-app.use(express.json())
+const mongoose = require("mongoose");
+const uri =
+  "mongodb+srv://alicjaxpoltorak:master3@cluster0.zaf91zr.mongodb.net/";
```
To powinno być w .env nie tutaj

KamilMr commented on models/contacts.js:
```diff
@@ -1,19 +1,65 @@
 
-const listContacts = async () => {}
+const listContacts = async () => {
+  const data = await fs.readFile(contactsPath);
```
Tutaj nadal używasz pliku, a powinnaś mieć schema contacts i wszystko tam zapisywac.

KamilMr commented on routes/api/users.js:
```diff
@@ -0,0 +1,92 @@
+
+  jwt.verify(token, process.env.ACCESS_TOKEN_SECRET, (err, user) => {
+    if (err) return res.status(403).json({ message: 'Invalid token' });
+    req.user = user;
```
A skąd masz user? Nie widzę zebyś go pobierała?

KamilMr replied:
Chodzi o to, że użytkownik jest zapisany w bazie danych. Tu musisz pobrać user z bazy danych, sprawdzić jego token. I wtedy go przypisać. Ma sens? :)

KamilMr commented on routes/api/users.js:
```diff
@@ -0,0 +1,92 @@
+
+router.get("/current", authenticateToken, (req, res) => {
+  res.status(200).json({
+    email: req.user.email,
```
zastanawiam się czy to przetestowałaś? Bo widzę że masz napisane, ale w authenticateToken nie widzę żebyś pobierała dane użytkownika tylko masz jego id.

KamilMr commented on routes/auth.js:
```diff
@@ -0,0 +1,19 @@
+const jwt = require('jsonwebtoken');
+const User = require('../models/User');
+
+module.exports = async (req, res, next) => {
```
To uzyj jako sprawdzające middleware a nie authenticateToken. Importuj go i uzyj. :)

KamilMr commented on routes/auth.js:
```diff
@@ -0,0 +1,19 @@
+
+    const user = await User.findOne({ _id: decoded.id, token });
+    if (!user) {
+      throw new Error();
```
Mozesz zwrócic not authorized też i tutaj :)

KamilMr reviewed (changes requested):
Hej, Kilka komentarzy w kodzie dodałem odnoścnie autoryzacji. 
1. Musisz jeszcze zacząć używać contacts db zamiast pliku i zapisywać kontakty z kluczem owner i odczytywać je. Czyli to co zrobiłaś z user tylko teraz dla kontakt (to z zeszłego zadania) 

2. A zebym zaakceptował to zadanie to ptrzebujesz poprawnie uzyc auth.

alicjapol committed:
json deleted

alicjapol committed:
Change API route path for contacts router

The route path for the contacts router has been changed from "/api/contacts" to "/routes/contacts".

alicjapol committed:
Remove Passport JWT configuration

The commit deletes the entire contents of the `passport.js` file, which contains the configuration for Passport JWT authentication middleware. It also removes the `secret` variable used to sign

alicjapol committed:
Update app.js

alicjapol committed:
updated login function
